frontend/src/app/pages/upload-page/upload-page.component.ts

```diff
@@ -264,13 +269,23 @@ export class UploadPageComponent implements OnInit, OnDestroy {
         const numberOfSlices = uploadingScan.scan.files.length;
 
         return this.scanService.createNewScan(dataset, numberOfSlices).then((scanId: string) => {
-                console.log('New Scan created with ID:', scanId, ', number of Slices:', numberOfSlices);
-                uploadingScan.id = scanId;
-                return this.scanService.uploadSlices(scanId, uploadingScan.scan.files);
-            },
-            () => {
-                return throwError({error: 'Could not create Scan.'});
+            console.log('New Scan created with ID:', scanId, ', number of Slices:', numberOfSlices);
+            uploadingScan.id = scanId;
+            uploadingScan.scan.predefinedLabels.forEach((predefinedLabel: SelectedScanPredefinedLabel) => {
```

Comment:
Is it safe here to assume that uploadingScan object has scan that has predefinedLabels collection?

Reply:
Yes, it is safe. This property is initialized to an empty array here:
https://github.com/jpowie01/MedTagger/blob/8058c2d99737564664f609d4509c7a334bf7335b/frontend/src/app/components/upload-scans-selector/upload-scans-selector.component.ts#L24